packages/core/src/interpreter.ts

```diff
@@ -1303,19 +1293,20 @@ export function spawn(
   const resolvedOptions = resolveSpawnOptions(nameOrOptions);
 
   return withServiceScope(undefined, (service) => {
+    const isLazyEntity = isMachine(entity) || isFunction(entity);
```

Comment:
this could be moved to the dev-only branch below so those calls would get removed in prod bundles